x-pack/plugins/integration_assistant/public/components/create_integration/create_integration_assistant/steps/cel_input_step/generation_modal.tsx

```diff
@@ -0,0 +1,222 @@
+      notifications?.toasts == null
+    ) {
+      return;
+    }
+    const generationStartedAt = Date.now();
+    const abortController = new AbortController();
+    const deps = { http, abortSignal: abortController.signal };
+
+    (async () => {
+      try {
+        const apiDefinition = integrationSettings.apiDefinition;
+        const celRequest: CelInputRequestBody = {
+          dataStreamName: integrationSettings.dataStreamName ?? '',
+          apiDefinition: apiDefinition ?? '',
+          connectorId: connector.id,
+          langSmithOptions: getLangSmithOptions(),
+        };
+        const celGraphResult = await runCelGraph(celRequest, deps);
+
+        if (abortController.signal.aborted) return;
+
+        if (isEmpty(celGraphResult?.results)) {
+          throw new Error('Results not found in response');
+        }
+
+        reportCelGenerationComplete({
+          connector,
+          integrationSettings,
+          durationMs: Date.now() - generationStartedAt,
+        });
+
+        const result = {
+          program: celGraphResult.results.program,
+          stateSettings: celGraphResult.results.stateSettings,
+          redactVars: celGraphResult.results.redactVars,
+        };
+
+        onComplete(result);
+      } catch (e) {
+        if (abortController.signal.aborted) return;
+        const errorMessage = `${e.message}${
+          e.body ? ` (${e.body.statusCode}): ${e.body.message}` : ''
+        }`;
+
+        reportCelGenerationComplete({
+          connector,
+          integrationSettings,
+          durationMs: Date.now() - generationStartedAt,
+          error: errorMessage,
+        });
+
+        setError(errorMessage);
+      } finally {
+        setIsRequesting(false);
+      }
+    })();
+    return () => {
+      abortController.abort();
+    };
+  }, [
+    isRequesting,
+    onComplete,
+    connector,
+    http,
+    integrationSettings,
+    reportCelGenerationComplete,
+    notifications?.toasts,
+  ]);
+
+  const retry = useCallback(() => {
+    setError(null);
+    setIsRequesting(true);
+  }, []);
+
+  return { error, retry };
+};
+
+const useModalCss = () => {
+  const { euiTheme } = useEuiTheme();
+  return {
+    headerCss: css`
+      justify-content: center;
+      margin-top: ${euiTheme.size.m};
+    `,
+    bodyCss: css`
+      padding: ${euiTheme.size.xxxxl};
+      min-width: 600px;
+    `,
+  };
+};
+
+interface GenerationModalProps {
+  integrationSettings: State['integrationSettings'];
+  connector: State['connector'];
+  onComplete: OnComplete;
+  onClose: () => void;
+}
+export const GenerationModal = React.memo<GenerationModalProps>(
+  ({ integrationSettings, connector, onComplete, onClose }) => {
+    const { headerCss, bodyCss } = useModalCss();
+    const { error, retry } = useGeneration({
+      integrationSettings,
+      connector,
+      onComplete,
+    });
+
+    return (
+      <EuiModal onClose={onClose} data-test-subj="generationModal">
```

Comment:
I hope this `data-test-subj` is unique value here , otherwise it can potentially break the cypress tests.

make sure to check all these values are unique

Reply:
I'll double check all these new ones. 

I don't think it would impact the cypress tests anyways though since this logic is all behind a feature flag, and doesn't get exposed otherwise.

Reply:
Yeah sure it should not impact, but when we want to write cypress tests for CEL flow , they might.